index.js

```diff
@@ -1,39 +1,44 @@
 (function(root) {
+  var docEl = document && document.documentElement;
   var scrollbarSize;
+  var scrollTop;
 
   function getScrollbarSize() {
     if (typeof scrollbarSize !== 'undefined') return scrollbarSize;
 
-    var doc = document.documentElement;
     var dummyScroller = document.createElement('div');
     dummyScroller.setAttribute('style', 'width:99px;height:99px;'
       + 'position:absolute;top:-9999px;overflow:scroll;');
-    doc.appendChild(dummyScroller);
+    docEl.appendChild(dummyScroller);
     scrollbarSize = dummyScroller.offsetWidth - dummyScroller.clientWidth;
-    doc.removeChild(dummyScroller);
+    docEl.removeChild(dummyScroller);
     return scrollbarSize;
   }
 
   function hasScrollbar() {
-    return document.documentElement.scrollHeight > window.innerHeight;
+    return docEl.scrollHeight > window.innerHeight;
   }
 
   function on(options) {
-    if (typeof document === 'undefined') return;
-    var doc = document.documentElement;
-    var rightPad = parseInt(getComputedStyle(doc)['padding-right'], 10);
-    doc.style.overflow = 'hidden';
+    if (!docEl) return;
+    scrollTop = window.pageYOffset;
     if (hasScrollbar()) {
-      rightPad += getScrollbarSize();
-      doc.style.paddingRight = rightPad + 'px';
+      docEl.style.width = 'calc(100% - '+ getScrollbarSize() +'px)';
```

Comment:
Yeah, browser support will be the same. I'm fine with it. In the future it's a good idea to separate completely independent ideas like this into different PRs/commits.